shaderc-rs/src/lib.rs

```diff
@@ -80,62 +80,87 @@ use std::{error, fmt, ptr, result, slice, str};
 
 /// Error.
 ///
-/// Each enumerants has an affixed string describing detailed reasons for
-/// the error. The string can be empty in cases.
+/// Each enumerant has an affixed string describing detailed reasons for
+/// the error. The string can be empty in certain cases.
 #[derive(Debug, PartialEq)]
 pub enum Error {
     /// Compilation error.
     ///
-    /// Contains the number of errors and detailed error string.
+    /// Contains the number of errors and a detailed error string.
     CompilationError(u32, String),
     InternalError(String),
     InvalidStage(String),
     InvalidAssembly(String),
     NullResultObject(String),
+
+    /// Returned when an internal initialization fails, for instance creating
+    /// a `Compiler` or `CompileOptions` fails because the underlying shaderc
+    /// function returned null.
+    InitializationError(String),
+
+    /// Returned when string parsing fails (e.g., version/profile parsing).
+    ParseError(String),
 }
 
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match *self {
-            Error::CompilationError(c, ref r) => {
-                if c == 1 {
+            Error::CompilationError(count, ref reason) => {
+                if count == 1 {
                     write!(f, "compilation error")?;
                 } else {
-                    write!(f, "{c} compilation errors")?;
+                    write!(f, "{} compilation errors", count)?;
                 }
-
-                if !r.is_empty() {
-                    write!(f, ":{}{}", if r.contains('\n') { "\n" } else { " " }, r)?;
+                if !reason.is_empty() {
+                    write!(
+                        f,
+                        ":{}{}",
+                        if reason.contains('\n') { "\n" } else { " " },
+                        reason
+                    )?;
                 }
-
                 Ok(())
             }
             Error::InternalError(ref r) => {
                 if r.is_empty() {
                     write!(f, "internal error")
                 } else {
-                    write!(f, "internal error: {r}")
+                    write!(f, "internal error: {}", r)
```

Comment:
these changes will make clippy sad

Reply:
@filnet thank you for pointing out! I've gone through these including `format!` etc. I hope it is better now. Please LMK if anything else would make it more suitable!